packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -238,12 +231,13 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
      * other GC nodes.
      * Derived classes must override this to provide custom list of references to other GC nodes.
      */
-    protected getGCDataCore(): IGarbageCollectionData {
+    protected async getGCDataCore(): Promise<IGarbageCollectionData> {
         // We run the full summarize logic to get the list of outbound routes from this object. This is a little
-        // expensive but its okay for now. It will be updated to not use full summarize and make it more efficient.
+        // expensive but it's okay for now. It will be updated to not use full summarize and make it more efficient.
         // See: https://github.com/microsoft/FluidFramework/issues/4547
         const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
-        this.snapshotCore(serializer);
+        const capture = this.captureSummaryStateCore(serializer, false);
+        await this.summarizeStateCore(serializer, capture);
```

Comment:
Is it okay to prevent use of `serializer` during this async call?

Reply:
I'm not sure what you mean. But looking at this function, it feels like it does not care about reentrancy at all (same for getGCData), so maybe we remove _isGCing checks?

Reply:
I meant the serializer getter, which we don't want used during GC. My concern is that making this async might allow some other code [not related to GC process] to run in the meantime which legitimately calls the serializer getter.